Sketch.java

```diff
@@ -1,36 +1,137 @@
 import processing.core.PApplet;
+import processing.core.PImage;
+import java.util.Set;
+import java.util.HashSet;
+
 
 public class Sketch extends PApplet {
 	
-	
+  Snow[] circles;
+  Player player;
+  public static final int SIZE = 400;
+  int boost = 0;
+  Set<Snow> hitSnowBalls = new HashSet<>();
+  
   /**
    * Called once at the beginning of execution, put your size all in this method
    */
   public void settings() {
-	// put your size call here
-    size(400, 400);
+	  // put your size call here
+    size(SIZE, SIZE);
   }
 
   /** 
    * Called once at the beginning of execution.  Add initial set up
    * values here i.e background, stroke, fill etc.
    */
   public void setup() {
-    background(210, 255, 173);
+    // set up five snowballs
+    circles = new Snow[5];
+    for (int i = 0; i < 5; ++i) {
+      circles[i] = new Snow();
+    }
+
+    // set up one player
+    player = new Player();
   }
 
   /**
-   * Called repeatedly, anything drawn to the screen goes here
-   */
+  * Returns true if x1, y1 and x2, y2 are close to each other (i.e. a hit)
+  */
+  public boolean isHit(int x1, int y1, int x2, int y2) {
+    return ((Math.abs(x1 - x2) < 13) &&
+            (Math.abs(y1 - y2) < 13));  
+  }
+
+  /**
+  * This function draws the screen.
+  */
   public void draw() {
-	  
-	// sample code, delete this stuff
-    stroke(128);
-    line(150, 25, 270, 350);  
+    background(50);
+
+    fill(255, 255, 255);
+    for (int i = 0; i < 5; ++i) {
+      if (!circles[i].ballHideStatus) {
+        // reduce lives when hit by snow
+        if (isHit(circles[i].x, circles[i].y, player.x, player.y)) {
+          hitSnowBalls.add(circles[i]);
+        }
+
+        // render snow ball
+        ellipse(circles[i].x, circles[i].y, 25, 25);
+        circles[i].y += circles[i].speed;
+        circles[i].y += boost;
+
+        // reset snowball once it moves off the screen
+        if (circles[i].y > height || circles[i].y < 0) {
+          circles[i] = new Snow();
+        }
+      }
+    }
 
-    stroke(255);
-    line(50, 125, 70, 50);  
+    // draws the player
+    fill(0, 0, 255);
+    ellipse(player.x, player.y, 20, 20);
+
+    // Draw the score squares
+    fill(255, 0, 0);
+    if (hitSnowBalls.size() < 1) {
+      rect(width - 90, 20, 20, 20);
+    }
+    if (hitSnowBalls.size() < 2) {
+      rect(width - 60, 20, 20, 20);      
+    }
+    if (hitSnowBalls.size() < 3) {
+      rect(width - 30, 20, 20, 20);      
+    }
+
+    // Complete the game when hit 3 times
+    if (hitSnowBalls.size() >= 3) {
+      background(255);
+    }
   }
-  
   // define other methods down here.
+
+  /**
+  * This function defines all the recognized keystrokes
+  **/
+  public void keyPressed() {
+
+    // up and down controls the speed of snowballs
+    if (key == CODED) {
+      if (keyCode == UP) {
+        ++boost;
+      } else if (keyCode == DOWN) {
+        --boost;
```

Comment:
this results in the balls going upward.  the intention is to keep falling down, but can go faster or slower based on up/down arrow.